### DISCUSSION TITLE
Performance Issues with 2k+ nodes (bubbles).

### DISCUSSION THREAD

**Discussion by HeinzContiZacherl** [foambubble/foam#1342] [General]
Hey i found your amazing work yesterday and it looks like a dream come true. Sadly im not being able to use it. This is happening:

Environment: MacBook Air M2; VScode; foam

1. I have a markdown document with 8000 lines. Its a very long one.
2. The document has around 2k wikilinks. The wikilinks where created in obsidian using the format `[[word]]`.
3. Half of those wikilinks are placeholders.
4. Foam makes than vscode really laggy and
5. when opening the graph it opens blank and the plugins host crashes and has to be restarted.

Would there be any solution or workaround for this? What could be causing this?

**Comment by riccardoferretti** [comment]
Thanks for reporting the issue, definitely a performance problem, which off the top of my head is given by both the size of the 8k lines md files (which stresses the parsing) and the 4k links (which I am guessing stretches the webview and the communication between that and the host process).
I would convert this into an issue, and if there is any way you could generate a repo that reproduces it, that'd be great.
Also, look into the existing performance related issues.

As workaround I would start by splitting the 8k lines files if it makes sense for your use case

**Comment by stereobooster** [comment]
I wonder if https://github.com/stereobooster/braindb can handle this case. It doesn't store data in the memory - instead it stores it in SQLite. And parsed documents are cached in SQLite as well - unless document changed it won't re-parse it

**Comment by riccardoferretti** [comment]
Hi @stereobooster I glanced at the project and it looks pretty interesting. In fact Foam is also aiming to achieve something along those lines, and eventually allow to use it as its own library (we used to have a package called `foam-core`, which now has been embedded for DX reasons in `foam-vscode` - you can see it in the [core directory](https://github.com/foambubble/foam/tree/master/packages/foam-vscode/src/core).

There were a few concepts in braindb that are pretty interesting, would love to investigate more and maybe exchange thoughts on it

**Reply by stereobooster** [reply]
I will ping you in Discord (haven't used it before)

**Comment by TomLucidor** [comment]
What are the cause of this, too many hyperlinks? Or just LOC?
